tests/test_installer.py

```diff
@@ -36,3 +43,194 @@ def test_ensure_admins(tljh_dir, admins, expected_config):
 
     # verify the list was flattened
     assert config["users"]["admin"] == expected_config
+
+
+def setup_conda(distro, version, prefix):
+    """Install mambaforge or miniconda in a prefix"""
+    if distro == "mambaforge":
+        installer_url, _ = installer._mambaforge_url(version)
+    elif distro == "miniforge":
+        installer_url, _ = installer._mambaforge_url(version)
+        installer_url = installer_url.replace("Mambaforge", "Miniforge3")
+    elif distro == "miniconda":
+        arch = os.uname().machine
+        installer_url = (
+            f"https://repo.anaconda.com/miniconda/Miniconda3-{version}-Linux-{arch}.sh"
+        )
+    else:
+        raise ValueError(
+            f"{distro=} must be 'miniconda' or 'mambaforge' or 'miniforge'"
+        )
+    with conda.download_miniconda_installer(installer_url, None) as installer_path:
+        conda.install_miniconda(installer_path, str(prefix))
+    # avoid auto-updating conda when we install other packages
+    run(
+        [
+            str(prefix / "bin/conda"),
+            "config",
+            "--system",
+            "--set",
+            "auto_update_conda",
+            "false",
+        ],
+        input="",
+        check=True,
+    )
+
+
+@pytest.fixture
+def user_env_prefix(tmp_path):
+    user_env_prefix = tmp_path / "user_env"
+    with mock.patch.object(installer, "USER_ENV_PREFIX", str(user_env_prefix)):
+        yield user_env_prefix
+
+
+def _specifier(version):
+    """Convert version string to SpecifierSet
+
+    If just a version number, add == to make it a specifier
+
+    Any missing fields are replaced with .*
+
+    If it's already a specifier string, pass it directly to SpecifierSet
+
+    e.g.
+
+    - 3.7 -> ==3.7.*
+    - 1.2.3 -> ==1.2.3
+    """
+    if version[0].isdigit():
+        # it's a version number, not a specifier
+        if version.count(".") < 2:
+            # pad missing fields
+            version += ".*"
+        version = f"=={version}"
+    return SpecifierSet(version)
+
+
+@pytest.mark.parametrize(
+    "distro, distro_version, expected_versions",
+    [
+        # No previous install, start fresh
+        (
+            None,
+            None,
+            {
+                "python": "3.10.*",
+                "conda": "23.1.0",
+                "mamba": "1.4.1",
+            },
+        ),
+        # previous install, 1.0
+        (
+            "mambaforge",
+            "22.11.1-4",
```

Comment:
I forgot this in the patch I committed @minrk!

```suggestion
            "23.1.0-1",
```

Reply:
Since tests takes quite a wile, I figure I'll commit this to reduce context switching.